### PR TITLE
basic project configuration

### PR DESCRIPTION
**Pull request by Javitocatral**

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -39,7 +40,7 @@ module.exports = {
     "no-console": ["warn", { allow: ["warn", "error"] }],
     "@typescript-eslint/no-unused-vars": [
       "error",
-      { argsIgnorePattern: "^_", ignoreRestSiblings: true },
+      // { argsIgnorePattern: '^_', ignoreRestSiblings: true },
```
Comment (zwergius) [inline .eslintrc.cjs:43]: this should be enabled

File: package.json [hunk 2]
```diff
@@ -8,21 +8,30 @@
...
+    "eslint-plugin-import": "^2.27.5",
+    "eslint-plugin-n": "^15.7.0",
+    "eslint-plugin-promise": "^6.1.1",
     "husky": "^8.0.3",
-    "lint-staged": "^13.2.1",
+    "lint-staged": "^13.2.2",
+    "postcss": "^8.4.23",
     "prettier": "2.8.7",
+    "prettier-plugin-svelte": "^2.10.0",
+    "stylelint-config-prettier": "^9.0.5",
```
Comment (zwergius) [inline package.json:30]: I think you can remove this

File: package.json [hunk 3]
```diff
@@ -32,9 +41,12 @@
...
   "lint-staged": {
-    "*.{js,ts,svelte}": "eslint"
+    "*.{js,ts,svelte}": [
+      "npm run check --",
+      "npm run lint --"
+    ]
   },
   "dependencies": {
-    "postcss": "^8.4.23"
+    "eslint-plugin-svelte3": "^4.0.0"
```
Comment (zwergius) [inline package.json:50]: move to dev dependencies

**Review by zwergius — CHANGES_REQUESTED** [review]

File: svelte.config.js [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
+  ],
 
   kit: {
     // adapter-auto only supports some environments, see https://kit.svelte.dev/docs/adapter-auto for a list.
     // If your environment is not supported or you settled on a specific environment, switch out the adapter.
     // See https://kit.svelte.dev/docs/adapters for more information about adapters.
     adapter: adapter(),
   },
-}
+};
```
Comment (zwergius) [inline svelte.config.js:22]: why the semi colons?

**Review by zwergius — CHANGES_REQUESTED** [review]

File: svelte.config.js (hunk 4 above)
Comment (Javitocatral) [reply svelte.config.js:22]: they get lonely

Comment (zwergius) [reply svelte.config.js:22]: But somehow you prettier is not working as it is not picking up your config file

**Review by zwergius — APPROVED** [review]

**Review by zwergius — CHANGES_REQUESTED** [review]

**Review by zwergius — APPROVED** [review]
LGTM
ok usa

**Review by zwergius — APPROVED** [review]

**Pull request merged**